File: ProjectCinemaSecurityFront/src/app/services/review.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { Review } from '../models/review.model';

@Injectable({
  providedIn: 'root'
})
export class ReviewService {

  constructor(private http: HttpClient) { }

  getReviewById(id: number): Observable<Review> {
    return this.http.get<Review>(`${environment.apiUrl}/Review/${id}`);
  }

  getReviewsByIdFilm(idFilm: number) {
    return this.http.get<Review[]>(`${environment.apiUrl}/Review/Film/${idFilm}`);
  }

  getAllReviews() {
    return this.http.get<Review[]>(`${environment.apiUrl}/Review`);
  }

  addReview(Review: Review) {
    return this.http.post(`${environment.apiUrl}/Review`, Review);
  }

  updateReview(Review: Review) {
    return this.http.put(`${environment.apiUrl}/Review`, Review);
  }

  delete(id: number) {
    return this.http.delete(`${environment.apiUrl}/Review/${id}`);
  }
}

File: ProjectCinemaSecurityBack/Services/ReviewService.cs
using ProjectCinemaSecurityBack.Context;
using ProjectCinemaSecurityBack.Models;
using ProjectCinemaSecurityBack.Repositories;

namespace ProjectCinemaSecurityBack.Services
{
    public class ReviewService
    {
        private ReviewRepository reviewRepository;
        public ReviewService(ReviewRepository repo)
        {
            reviewRepository = repo;
        }

        public ReviewModel AddReview(ReviewModel review)
        {
            return this.reviewRepository.AddReview(review);
        }

        public void DeleteReview(int id)
        {
            this.reviewRepository.DeleteReview(id);
        }

        public ReviewModel GetReviewById(int id)
        {
            return this.reviewRepository.GetReviewById(id);
        }

        public IEnumerable<ReviewModel> GetReviewsByIdFilm(int idFilm)
        {
            return this.reviewRepository.GetReviewsByIdFilm(idFilm);
        }

        public IEnumerable<ReviewModel> GetReviews()
        {
            return this.reviewRepository.GetReviews();
        }

        public ReviewModel UpdateReview(ReviewModel review)
        {
            return this.reviewRepository.UpdateReview(review);
        }
    }
}
